cbor2/decoder.py

```diff
@@ -367,7 +357,7 @@ def decode(self, shareable_index=None):
             return decoder(self, subtype, shareable_index)
         except CBORDecodeError:
             raise
-        except Exception as e:
+        except Exception as e:  # pragma: no cover
```

Comment:
This pragma should only be used for inherently untestable code, not circumventing code coverage limits.

Reply:
The question here is, is this case reproduceable? If not, why?

Reply:
I think it might be testable, I'll keep poking it with a stick.

Reply:
Okeydoke, I can test it like this in python 3:

```python
def test_nested_exception():
    import sys
    if sys.version_info.major >= 3:
        exc = pytest.raises((CBORDecodeError, TypeError), loads, unhexlify('A1D9177082010201'))
        exc.match("error decoding value at index 8: unhashable type: 'CBORTag'")
    else:
        assert True
```

Any objections?

Reply:
Hm, does this really only happen on py3? If so, you should skip the test on py2 (see the pytest documentation on how to do that with a decorator).

Reply:
Yep I did this in 
https://github.com/agronholm/cbor2/pull/13/commits/c17586f440f816465bb7542d8ff53cf26b5c3d61